graveler/committed/unmarshal_iterator.go

```diff
@@ -34,7 +34,7 @@ func (r *UnmarshalIterator) Next() bool {
 		}
 	}
 	r.value = &graveler.ValueRecord{
-		Key:   graveler.Key(val.Key),
+		Key:   graveler.Key(val.Key.Copy()),
```

Comment:
This is something of an opportunity to get some benefit from what little type safety Go gives: if we wrote this as a function that copies and converts, it would be easier and lazier to do the right thing.